### DISCUSSION TITLE
Customizing reports

### DISCUSSION THREAD

**Discussion by christianlehre** [google/meridian#293] [Q&A]
Cheers. How easily customizable are the reports, e.g. the summary report? I really like how the report-generation is abstracted away from the users, but I would like to customize it, e.g. by plotting the model fit for each geo in my hierarchical model, plotting separate response curves and/or including adstock and decay curves.

**Comment by brendalewisprice** [comment]
Glad to hear you are enjoying the html report generation! 

Currently the only customizable feature for the html output files is the date range (start and end dates), however each of the plotted elements can be created independently from the underlying data or by using methods from the `visualizer` module.

Most of the data used to generate the summary report can be found in the analyzer module. For example, `Analyzer.predictive_accuracy` returns metrics for predictive accuracy, including R-squared, MAPE, wMAPE. Some properties allow customization for geo and time (see [`analyzer.py`](https://github.com/google/meridian/blob/main/meridian/analysis/analyzer.py) for details on each). 

To specifically address you question about separate response curves, we do have that functionality: you can use the `plot_separately` argument in the `plot_response_curves `method in the `visualizer` module, or you can directly access the data used to create the response curves in `Analyzer.response_curves`.
